core/crypto/aes-gcm-sw.c

```diff
@@ -43,6 +43,8 @@ void internal_aes_gcm_ghash_update(struct internal_aes_gcm_state *state,
 	if (head)
 		ghash_update_block(state, head);
 
+	assert((num_blocks == 0) || (data != NULL));
```

Comment:
I'm not sure this is really needed, but not a strong opinion.
This may be useful only for those who hack into AES GCM software implementation.

Note: with optee_os coding style, this should rather be `assert(!num_blocks || data);`.

Reply:
This looks like a random place to put this assert. Why is this needed?

Reply:
assert to make sure internal_aes_gcm_xor_block would not end up dereferencing data that could have been NULL

Reply:
I get that part. Why is it needed? What prompted this?

Reply:
Be caution: `assert()` has effects only in debug mode (`CFG_TEE_CORE_DEBUG=y` in optee_os).
Faulting assert will make Core to panic with a panic trace.
Dereferencing a NULL pointer will make Core to panic with a data abort trace.
In the end, same result.

`assert()` is only useful for debugging purposing to get an explicit trace on an specific unexpected event. Note that there are friendly backtrace tools in optee_os to analyse a data abort for example.

Reply:
cpp check highlighted the issue of pointer arithmetic on potential null pointer. Id be open to other methods of hardening internal_aes_gcm_ghash_update...

Reply:
changed to optee coding style